pygments/lexers/q.py

```diff
@@ -0,0 +1,228 @@
+                    + ("select", "setenv", "sin", "sqrt", "ss", "sum")
+                    + ("tan", "update", "var")
+                    + ("wavg", "while", "within", "wsum", "xexp"),
+                    suffix=r"\b",
+                ),
+                Keyword,
+            ),
+        ],
+        "declarations": [
+            (r"^\\t[s]?", Comment.Preproc),
+            (
+                r"^(\\\w\s+[^/\n]*?)([/].*)",
+                bygroups(Comment.Preproc, Comment.Single),
+            ),
+            (r"^\\\w+.*$", Comment.Preproc),
+            # Prompt
+            (r"^[a-zA-Z]\)", Generic.Prompt),
+            # Function Names
+            (
+                r"([.]?[a-zA-Z](?:\w|[.])*)(\s*)([-.~=!@#$%^&*_+|,<>?/\\:']?:)\s*(\{)",
+                bygroups(Name.Function, Whitespace, Operator, Punctuation),
+                "functions",
+            ),
+            # Variable Names
+            (
+                r"([.]?[a-zA-Z](?:\w|[.])*)(\s*)([-.~=!@#$%^&*_+|,<>?/\\:']?:)",
+                bygroups(Name.Variable, Whitespace, Operator),
+            ),
+            # Functions
+            (r"\{", Punctuation, "functions"),
+            # Parentheses
+            (r"\(", Punctuation, "parentheses"),
+            # Brackets
+            (r"\[", Punctuation, "brackets"),
+            # Errors
+            (r"'`[a-zA-Z]*(\w|[.])*", Name.Exception),
+            # File Symbols
+            (r"`:[a-zA-Z/]*(\w|[./])*", String.Symbol),
+            # Symbols
+            (r"[`][a-zA-Z]*(\w|[.])*", String.Symbol),
+            # Numbers
+            include("numbers"),
+            # Variable Names
+            (r"[a-zA-Z](\w|[.])*", Name),
+            # Operators
+            (r"[-=+*#$%@!~^&:.,<>'\\|/?_]", Operator),
+            # Punctuation
+            (r"[;]", Punctuation),
+        ],
+        "functions": [
+            include("root"),
+            (r"\}", Punctuation, "#pop"),
+        ],
+        "parentheses": [
+            include("root"),
+            (r"\)", Punctuation, "#pop"),
+        ],
+        "brackets": [
+            include("root"),
+            (r"\]", Punctuation, "#pop"),
+        ],
+        "numbers": [
+            (r"[01]+b", Number.Bin),
+            (r"0[nNwW][cefghijmndzuvtp]?", Number),  # nulls/infinities
+            (
+                (
+                    r"(?:[0-9]{4}[.][0-9]{2}[.][0-9]{2}|[0-9]+)"
+                    "D(?:[0-9](?:[0-9](?:[:][0-9]{2}"
+                    "(?:[:][0-9]{2}(?:[.][0-9]*)?)?)?)?)?"
+                ),
+                Literal.Date,  # timestamp
+            ),
+            (
+                (
+                    r"[0-9]{4}[.][0-9]{2}"
+                    "(?:m|[.][0-9]{2}(?:T(?:[0-9]{2}:[0-9]{2}"
+                    "(?:[:][0-9]{2}(?:[.][0-9]*)?)?)?)?)"
+                ),
+                Literal.Date,  # datetime
+            ),
+            (
+                (r"[0-9]{2}:[0-9]{2}" "(?:[:][0-9]{2}" "(?:[.][0-9]{1,3})?)?"),
```

Comment:
The implicit string concatenation on the same line seems coming from a reformatting.

Reply:
yes, this was needed to allow the string to spread across lines. i subsequently optimized the regex and the black formatter merged the shortened lines.  will fix.